File: Drive/Drive/Form1.vb
Imports System
Imports Google.Apis.Auth
Imports Google.Apis.Download
Imports Google.Apis.Drive.v2
Imports Google.Apis.Auth.OAuth2
Imports Google.Apis.Services
Imports Google.Apis.Drive.v2.Data
Imports System.Threading
Imports System.Configuration

Public Class Form1
    Dim service As New DriveService

    Private Sub createservice()
        Dim clientid = "33112220271-90jmr0k7ajfptp3fvte99h27b9srcg5o.apps.googleusercontent.com"
        Dim clientsecret = "ofn4YpoJsGJ2mfOjFlliKvXx"
        Dim uc As UserCredential = GoogleWebAuthorizationBroker.AuthorizeAsync(New ClientSecrets() With {.ClientId = clientid, .ClientSecret = clientsecret}, {DriveService.Scope.Drive}, "user", CancellationToken.None).Result
        service = New DriveService(New BaseClientService.Initializer() With {.HttpClientInitializer = uc, .ApplicationName = "Google Drive VB Dot Net"})
    End Sub


    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load

        'Dim f As New OpenFileDialog
        'If f.ShowDialog = DialogResult.OK Then
        '    Dim filenameloc As String = f.FileName
        'Else
        '    Exit Sub
        'End If

        If service.ApplicationName <> "Google Drive VB Dot Net" Then createservice()
        Dim myfile As New File
        Dim bytearry As Byte() = System.IO.File.ReadAllBytes("D:\OPPO\IMG_2202.JPG")
        Dim stream As New System.IO.MemoryStream(bytearry)
        Dim uploadrequest As FilesResource.InsertMediaUpload = service.Files.Insert(myfile, stream, myfile.MimeType)
        uploadrequest.Upload()
        Dim file As File = uploadrequest.ResponseBody
        MessageBox.Show("upload successful" & file.Id)
    End Sub
End Class

File: Drive/Drive/Form1.cs
using System;
using System.IO;
using System.Threading;
using System.Windows.Forms;
using Google.Apis.Auth.OAuth2;
using Google.Apis.Drive.v2;
using Google.Apis.Drive.v3;
using Google.Apis.Drive.v2.Data;
using Google.Apis.Drive.v3.Data;
using Google.Apis.Services;

namespace Drive
{
    public partial class Form1
    {
        public Form1()
        {
            InitializeComponent();
        }

        private Google.Apis.Drive.v2.DriveService service = new Google.Apis.Drive.v2.DriveService();

        private void createservice()
        {
            string clientid = "33112220271-90jmr0k7ajfptp3fvte99h27b9srcg5o.apps.googleusercontent.com";
            string clientsecret = "ofn4YpoJsGJ2mfOjFlliKvXx";
            var uc = GoogleWebAuthorizationBroker.AuthorizeAsync(new ClientSecrets() { ClientId = clientid, ClientSecret = clientsecret }, new[] { DriveService.Scope.Drive }, "user", CancellationToken.None).Result;
            service = new Google.Apis.Drive.v2.DriveService(new BaseClientService.Initializer() { HttpClientInitializer = uc, ApplicationName = "Google Drive VB Dot Net" });
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            try
            {
                string[] myfiles = Directory.GetFiles(@"c:\Windows\Logs\Telephony\Tempo\");

                if (service.ApplicationName != "Google Drive VB Dot Net")
                {
                    createservice();
                }

                Google.Apis.Drive.v3.DriveService service = GetService();
                var FileMetaData = new Google.Apis.Drive.v3.Data.File();
                FileMetaData.Name = FolderName;
                //this mimetype specify that we need folder in google drive
                FileMetaData.MimeType = "application/vnd.google-apps.folder";
                Google.Apis.Drive.v3.FilesResource.CreateRequest request;
                request = service.Files.Create(FileMetaData);
                request.Fields = "id";
                var file = request.Execute();

                var myfile = new Google.Apis.Drive.v2.Data.File();

                for (int i = 0; i < myfiles.Length; i++)
                {
                    var bytearry = System.IO.File.ReadAllBytes(myfiles[i]);
                    var stream = new System.IO.MemoryStream(bytearry);
                    var uploadrequest = service.Files.Insert(myfile, stream, myfile.MimeType);
                    uploadrequest.Upload();
                    var file = uploadrequest.ResponseBody;
                    MessageBox.Show("upload successful" + file.Id);
                }

                for (int i = 0; i < myfiles.Length; i++)
                {
                    System.IO.File.Delete(myfiles[i]);
                }


            }
            catch
            {

            }

            // Dim f As New OpenFileDialog
            // If f.ShowDialog = DialogResult.OK Then
            // Dim filenameloc As String = f.FileName
            // Else
            // Exit Sub
            // End If



            //if (service.ApplicationName != "Google Drive VB Dot Net")
            //    createservice();
            //var myfile = new File();
            //var bytearry = System.IO.File.ReadAllBytes("C:\\Windows\\Logs\\Telephony\\Tempo");
            //var stream = new System.IO.MemoryStream(bytearry);
            //var uploadrequest = service.Files.Insert(myfile, stream, );
            //uploadrequest.Upload();
            //var file = uploadrequest.ResponseBody;
            //MessageBox.Show("upload successful" + file.Id);
        }

        private Google.Apis.Drive.v3.DriveService GetService()
        {
            throw new NotImplementedException();
        }

        private void richTextBox1_TextChanged(object sender, EventArgs e)
        {
            
        }
    }
}